### PR TITLE
Create extending.md

### PR DESCRIPTION
**Pull request by manez**
**GitHub Issue**: https://github.com/Islandora/documentation/issues/1040

# What does this Pull Request do?

Adds a new page of documentation describing and linking out to the Cookbook and other resources. Brief, but I think it's worthwhile to point from the official docs to these other resources.

# How should this be tested?

Preview & read through. Test links. 

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

# Interested parties
@Islandora/8-x-committers or anyone else who wants to [sign up](https://docs.google.com/spreadsheets/d/1E-kRw9xE60CKK0qL1-phzeVKjEZu3qBKZ9d3LH1hDEE) to review!

### CODE REVIEW

**Review by whikloj — APPROVED** [review]
👍

**Comment by whikloj** [comment]
As this is a new page, you need add it to the `mkdocs.yml` file or it doesn't appear in the menu. 

You'll want to add a line in the appropriate position in this block
https://github.com/Islandora/documentation/blob/master/mkdocs.yml#L39-L53

**Comment by manez** [comment]
@whikloj added to the mkdocs file!

File: docs/user-documentation/extending.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Extending Islandora
+ 
+Because Islandora 8 is very tightly integrated with Drupal, many use cases that required custom Islandora modules in previous versions can now be solved with a combination of configuration and modules from the Drupal community. Other use cases may still require custom code, but are not common enough, or complete enough, for that custom code to be a part of the core software. 
+ 
+Both scenarios are covered by the [Islandora Cookbook](https://github.com/Islandora-Labs/Islandora-Cookbook), which contains an annotated list of custom Islandora tools and modules developed by our community, a list of Druapl contributed modules that may be useful in an islandora context, and a set of _recipes_ which each provide a detailed outline for solving a particular use case.
```
Comment (librarychik) [inline docs/user-documentation/extending.md:5]: Spelling: Drupal

" ...a list of Druapl contributed modules ..."

**Review by librarychik — COMMENTED** [review]
Links work, one spelling error.  tagging @Islandora/8-x-committers

File: docs/user-documentation/extending.md (hunk 1 above)
Comment (manez) [reply docs/user-documentation/extending.md:5]: oops! Good catch, @librarychik. Any chance you could make a pull request to fix that? I have so many branches open spread across different pages that I can't "rebase" (delete my repo and refork 😉 ), and I can't seem to get a new commit against this one without dragging a bunch of cruft.

Comment (librarychik) [reply docs/user-documentation/extending.md:5]: Happy to help.  I can't see a way to do a PR, can you point me in the right direction?

Comment (manez) [reply docs/user-documentation/extending.md:5]: @librarychik to do it in the GUI:

1. Click "Fork" (up at the top of the page here) to make your own fork. 
2. Find the page in your fork (/documentation/blob/master/docs/user-documentation/extending.md) and edit in the browser.
3. Commit the changes to your fork.
4. Go back to the home page of that repo in your fork and click "new pull request." The template should make it clear from there.

Comment (librarychik) [reply docs/user-documentation/extending.md:5]: Okay, done.  (hopefully correctly 😄 )

Comment (manez) [reply docs/user-documentation/extending.md:5]: Awesome! https://github.com/Islandora/documentation/pull/1357 looks good to me. We'll need a Committer to swoop in and hit the merge button. @whikloj ?

Comment (seth-shaw-unlv) [reply docs/user-documentation/extending.md:5]: I have you covered.

**Pull request merged**